### ISSUE TITLE
Unable to use {forwardRef: true} option of observer HOC

### ISSUE THREAD

**Issue by leantide** [mobxjs/mobx-react#868]
## Intended outcome

I would like to forward a ref to an observed component as per my current understanding of https://mobx-react.js.org/observer-hoc

```
const MyObserverComponent = observer(
  (props, ref) => {
    return <div ref={ref}>My Observer Component</div>;
  },
  { forwardRef: true }
);
```

## Actual outcome

As per codesandbox below, React chokes:

Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Warning: Unexpected ref object provided for div. Use either a ref-setter function or React.createRef().

The above error occurred in the <div> component:
    in div (at App.js:29)
    in _c4 (at App.js:18)
    in div (at App.js:16)
    in App (at src/index.js:9)
    in StrictMode (at src/index.js:8)

## How to reproduce the issue

https://codesandbox.io/s/inspiring-colden-uettu?file=/src/App.js

## Versions

react: 16.12.0
mobx: 5.15.4
mobx-react: 6.2.2

**Comment by FredyC** [comment]
Thanks for the report. That option works only with `mobx-react-lite`. However, I don't see any reason why it shouldn't be here as well. I will leave this open and PR is welcome.

**Comment by leantide** [comment]
Sorry, I'm still trying to grasp the relationship between mobx-react and mobx-react-lite. I appreciate the latter is with hooks. 

Am I correct in saying that mobx-react-lite is its own library with a degree of independence from mobx-react and can be used in a hooks only implementation, but since mobx-react includes mobx-react-lite in its packaging, the mobx-react-lite API becomes part of the mobx-react API, but with exceptions (like this example) that may be in mobx-react-lite and not mobx-react but are documented in mobx-react for consistency?

**Comment by FredyC** [comment]
That is correct. Perhaps this overview will help you? https://mobx-react.js.org/libraries

The `observer` in `mobx-react` is slightly different because it needs to take care of class components too. The `forwardRef` option was added to `mobx-react-lite` much later out of specific need and nobody ever asked for it in `mobx-react`.

Actually, looking at the code, it seems that you should be able to wrap to `React.forwardRef` directly along with `observer` call.

**Comment by FredyC** [comment]
Yea, here is a quick demo.

https://codesandbox.io/s/xenodochial-wiles-3zvwk?file=/src/App.js

Such inconsistency is of course bad for any future migrations. I am thinking we should probably make it work the same way in `mobx-react-lite` instead of extra option.

**Comment by yuwanlin** [comment]
@FredyC thank you, It help me a lot.  [the demo](https://codesandbox.io/s/xenodochial-wiles-3zvwk?file=/src/App.js) is useful. 
for a function component, It seems
```
import {  inject } from 'mobx-react';
import { observer } from 'mobx-react-lite';

export default inject('commonStore')(observer(GamePage, {
  forwardRef: true
}));

```

and 

```
import {  observer, inject } from 'mobx-react';
export default inject('commonStore')(observer(React.forwardRef(GamePage)));
```
both work.

**Comment by MoonBall** [comment]
@FredyC From [code](https://github.com/mobxjs/mobx-react-lite/blob/master/src/observer.ts#L36), there is an intentional behavior for case under discussion.

**Comment by FredyC** [comment]
I think you are missing the point here. It should work the same way across both libraries.

**Comment by MoonBall** [comment]
> I think you are missing the point here. It should work the same way across both libraries.

@FredyC I mean the below code should not work because of [the code](https://github.com/mobxjs/mobx-react-lite/blob/master/src/observer.ts#L36). The comment of source code is for this case.

```
const MyObserverComponent = observer(
  (props, ref) => {
    return <div ref={ref}>My Observer Component</div>;
  },
  { forwardRef: true }
);
```

**Comment by yuwanlin** [comment]
@MoonBall [the code](https://github.com/mobxjs/mobx-react-lite/blob/master/src/observer.ts#L36) you mentioned is source code of `mobx-react-lite`... It work in this case. But `mobx-react` doesn't work in this case

**Comment by yuwanlin** [comment]
> Thanks for the report. That option works only with `mobx-react-lite`. However, I don't see any reason why it shouldn't be here as well. I will leave this open and PR is welcome.

I am very interested in fixing this problem.  Is there anything I need to pay attention to?

**Comment by yuwanlin** [comment]
> > Thanks for the report. That option works only with `mobx-react-lite`. However, I don't see any reason why it shouldn't be here as well. I will leave this open and PR is welcome.
> 
> I am very interested in fixing this problem. Is there anything I need to pay attention to?(eg: which version should I fix it?)

**Comment by FredyC** [comment]
@yuwanlin I am thinking it should work the other way around and expand `mobx-react-lite` to work with `React.forwardRef` directly instead of extra option. And possibly using that option could produce a one-time warning about deprecation and remove it from the next major. So if you are interested in that, go ahead.
